src/axom/inlet/Table.cpp

```diff
@@ -65,110 +65,88 @@ std::shared_ptr<Table> Table::addTable(const std::string& name,
 {
-  auto table = addTable(appendPrefix(name, "_inlet_array"), description);
-  std::unordered_map<int, int> map;
-  const std::string& fullName = appendPrefix(m_name, name);
-  if(m_reader->getIntMap(fullName, map))
-  {
-    for(auto p : map)
-    {
-      table->addIntHelper(std::to_string(p.first), "", true, p.second);
-    }
-  }
-  else
-  {
-    SLIC_WARNING(fmt::format("Int array {0} not found.", fullName));
-  }
-  return table;
+  return addPrimitiveArray<int>(name, description);
 }
 
-std::shared_ptr<Table> Table::addDoubleArray(const std::string& name,
-                                             const std::string& description)
+std::shared_ptr<Verifiable> Table::addDoubleArray(const std::string& name,
+                                                  const std::string& description)
 {
-  auto table = addTable(appendPrefix(name, "_inlet_array"), description);
-  std::unordered_map<int, double> map;
-  const std::string& fullName = appendPrefix(m_name, name);
-  if(m_reader->getDoubleMap(fullName, map))
-  {
-    for(auto p : map)
-    {
-      table->addDoubleHelper(std::to_string(p.first), "", true, p.second);
-    }
-  }
-  else
-  {
-    SLIC_WARNING(fmt::format("Double array {0} not found.", fullName));
-  }
-  return table;
+  return addPrimitiveArray<double>(name, description);
+}
+
+std::shared_ptr<Verifiable> Table::addStringArray(const std::string& name,
+                                                  const std::string& description)
+{
+  return addPrimitiveArray<std::string>(name, description);
 }
 
-std::shared_ptr<Table> Table::addStringArray(const std::string& name,
-                                             const std::string& description)
+std::shared_ptr<Table> Table::addGenericArray(const std::string& name,
+                                              const std::string& description)
 {
   auto table = addTable(appendPrefix(name, "_inlet_array"), description);
-  std::unordered_map<int, std::string> map;
+  std::vector<int> indices;
   const std::string& fullName = appendPrefix(m_name, name);
-  if(m_reader->getStringMap(fullName, map))
-  {
-    for(auto p : map)
+  if(m_reader->getArrayIndices(fullName, indices))
+  {
+    // This is how an array of user-defined type is differentiated
+    // from an array of primitives - the tables have to be allocated
+    // before they are populated as we don't know the schema of the
+    // generic type yet
+    auto view =
+      table->m_sidreGroup->createViewAndAllocate("_inlet_array_indices",
+                                                 axom::sidre::INT_ID,
+                                                 indices.size());
+    int* raw_array = view->getArray();
+    std::copy(indices.begin(), indices.end(), raw_array);
+    for(const auto idx : indices)
     {
-      table->addStringHelper(std::to_string(p.first), "", true, p.second);
+      table->addTable(std::to_string(idx), description);
```

Comment:
Is this the right description?

Reply:
That description is just the user description for the array of structs being used as the description for each element - it looks like it was previously a blank description, so I'm not sure which would be more helpful.

Reply:
Is there a way to get a description for each element?